_posts/2021-12-27-trino-2021-a-year-of-growth.md

```diff
@@ -0,0 +1,346 @@
+
+* Added support for reading fields as `json` values.
+* Fixed failure when documents contain fields of unsupported types.
+* Added support for `scaled_float` type.
+* Added support for assuming an IAM role.
+* Fixed potential incorrect results when queries contain an `IS NULL` predicate.
+* Added retry requests with backoff when Elasticsearch is overloaded.
+
+### Kafka connector
+
+* Added support for `TLS` security protocol.
+* Fixed querying of Schema Registry tables with References in their schema.
+* Fixed listing of schema registry tables having ambiguous subject name in lower case.
+
+### MemSQL connector
+
+* Allowed limiting the size of the metadata cache via the 
+`metadata.cache-maximum-size` configuration property.
+* Added partial support for `DELETE` statement where predicate can be fully pushed 
+down to the remote datasource.
+* Improved metadata caching hit rate.
+* Added support for MemSQL 3.2.
+
+### MongoDB connector
+
+* Added [`timestamp_objectid()`](https://trino.io/docs/current/connector/mongodb.html#timestamp_objectid)
+function.
+* Enabled `mongodb.socket-keep-alive` config property by default.
+* Add support for `json` type.
+* Support reading MongoDB `DBRef` type.
+* Allow skipping creation of an index for the `_schema` collection if it already
+exists.
+* Added support to redact the value of `mongodb.credentials` in the server log.
+* Added support for dropping columns.
+
+### MySQL connector
+
+* Added support for reading and writing `timestamp` values with precision higher than 3.
+* Added support for predicate pushdown on `timestamp` columns.
+* Exclude an internal `sys` schema from schema listings.
+
+### Pinot connector
+
+* Updated Pinot connector to be compatible with versions >= 0.8.0 and drop 
+support for older versions.
+* Added support for pushdown of filters on `varbinary` columns to Pinot.
+* Fixed incorrect results for queries that contain aggregations and `IN` and 
+`NOT IN` filters over varchar columns.
+* Fixed failure for queries with filters on `real` or `double` columns having 
+`+Infinity` or `-Infinity` values.
+* Implemented aggregation pushdown for `count`, `avg`, `min`, `max`, `sum`, 
+`count(DISTINCT)` and `approx_distinct`. It is enabled by default and can be 
+disabled using the configuration property `pinot.aggregation-pushdown.enabled` 
+or the catalog session property `aggregation_pushdown_enabled`.
+* Allowed https URLs in `pinot.controller-urls`. 
+
+## Runtime Improvements
+
+* Improved CPU coordinator utilization.
+* Improved query performance by reducing CPU overhead of repartitioning data 
+across worker nodes.
+* Reduced graceful shutdown time for worker nodes.
+
+## Everything else
+
+* Added support for ARM64 in the Trino Docker image.
+* Added cancelling a Trino query execution when JDBC statement is closed.
+* Added `clear` command to the Trino CLI to clear the screen.
+
+
+## Trino Summit
+
+This year as with the name change, we had the inaugural Trino Summit hosted by
```

Comment:
In 2021 we also enjoyed a successful inaugural Trino Summit, hosted by Starburst, with well over 500 attendees.